apps/dgst.c

```diff
@@ -467,9 +489,17 @@ int do_fp(BIO *out, unsigned char *buf, BIO *bp, int sep, int binout,
     if (binout) {
         BIO_write(out, buf, len);
     } else if (sep == 2) {
+        int backslash = 0;
+        file = newline_escape_filename(file, &backslash);
+
         for (i = 0; i < (int)len; i++)
-            BIO_printf(out, "%02x", buf[i]);
+            if (i == 0 && backslash == 1)
```

Comment:
Why??

Reply:
This adds the backslash as the first character.  If this is undesired, I can remove.

Reply:
I'm wondering what purpose this serves?
The newlines are escaped separately.

Reply:
This was originally based off of the coreutils format when displaying a digest summary using `sha256sum *`
With the output of:
```
\911169ddaaf146aff539f58c26c489af3b892dff0fe283c1c264c65ae5aa59a2  a\nb
e1118418fb157664605bcafd07c25befa2ebe52d001a16f63e2f487c8db79d02   ACKNOWLEDGEMENT
```
This has been removed in my new commit.

Reply:
Hmmm, I think that initial backslash serves a purpose. Could there be *some* maker to ensure that `\n` gets interpreted as a newline and not an actual backslash followed by 'n'?

Reply:
What about `\\` in addition the `\n`?
There is otherwise ambiguity if a filename has both \\ and \n in it.

Reply:
Yes, I was thinking about that, but that will break backward compatibility, as a backslash wasn't previously escaped...  so scripts that have relied on the current no-escape behaviour will interpret any double backslash as just that, and backslash followed by 'n' as backslash followed by 'n'.

I believe there's quite a lot of reasoning behind the choices made with `sha256sum`'s output format, and that we should probably listen...

Reply:
Thank you for this feedback.  I think the backward compatibility aspect presents a strong argument in adding the backslash back to the beginning of the digest. 

What are your thoughts on my previous implementation in adding the backslash to the beginning of the digest?

Reply:
I'm not entirely sure I like that format in particular. *However*, since that's already implemented in `*sum` (not just `sha256sum`), it already has recognition, and that makes it justifiable enough to replicate it, unless someone comes up with something better (I cannot).

So yeah, please add it back.

Reply:
Oh, and since this became a question with a lot of reasoning, it might be good to write a comment on these details, perhaps as a heading to the function?

Reply:
I think that is is great idea.  This will point out the reasoning behind this choice for anyone in the future looking at this code.  I will put together a new commit.